File: ayesha/Basics of Programming/Function-And-Arrays/any-base-multiplication.cpp
#include <iostream>
#include <math.h>
using namespace std;

int getSum(int b, int n1, int n2)
{
    // write your code here
    int sum = 0;
    int count = -1;
    int carry = 0;
    while (n1 != 0 || n2 != 0 || carry != 0)
    {
        count++;
        int ld1 = n1 % 10;
        int ld2 = n2 % 10;
        n1 = n1 / 10;
        n2 = n2 / 10;
        int ld = ld1 + ld2 + carry;
        carry = ld / b;
        ld = ld % b;
        int res = ld * pow(10, count);
        sum += res;
    }
    return sum;
}

int oneDigitProduct(int b, int n1, int ld2)
{
    int oneSum = 0;
    int count = -1;
    int carry = 0;
    while (n1 > 0 || carry > 0)
    {
        count++;
        int ld1 = n1 % 10;
        n1 = n1 / 10;
        int ld = ld1 * ld2 + carry;

        carry = ld / b;
        ld = ld % b;
        oneSum += ld * pow(10, count);
    }
    return oneSum;
}


int getProduct(int b, int n1, int n2)
{
    int sumAll = 0;
    int count = -1;
    while (n2 > 0)
    {
        count++;
        int ld2 = n2 % 10;
        n2 = n2 / 10;

        int all = oneDigitProduct(b, n1, ld2);
        sumAll = getSum(b, sumAll, all * pow(10, count));
    }
    return sumAll;
}

int main()
{
    int b, n1, n2;
    cin >> b >> n1 >> n2;
    cout << getProduct(b, n1, n2) << endl;
}

File: V.Vennela/Basics of Programming/Functions and Arrays/AnyBaseMultiplication.java
import java.util.*;

public class AnyBaseMultiplication{

public static void main(String[] args) {
    Scanner scn = new Scanner(System.in);
    int b = scn.nextInt();
    int n1 = scn.nextInt();
    int n2 = scn.nextInt();

    int d = getProduct(b, n1, n2);
    System.out.println(d);
 }

 public static int getProduct(int b, int n1, int n2){
     // write your code here
     int i=1;
       int a1=0;
       while(n1!=0)
       {
           int rem=n1%10;
           a1=a1+rem*i;
           i=i*b;
           n1=n1/10;
       }
        i=1;
       int a2=0;
       while(n2!=0)
       {
           int rem=n2%10;
           a2=a2+rem*i;
           i=i*b;
           n2=n2/10;
       }
       int n=a2*a1;
       i=1;
       int res=0;
       while(n!=0)
       {
           int rem=n%b;
           res=res+rem*i;
           i=i*10;
           n=n/b;
       }
       return res;
 }

}
